src/mark.js

```diff
@@ -78,9 +78,11 @@ export class Mark {
   initialize(facets, facetChannels) {
     let data = arrayify(this.data);
     if (facets === undefined && data != null) facets = [range(data)];
+    const originalFacets = facets;
     if (this.transform != null) ({facets, data} = this.transform(data, facets)), (data = arrayify(data));
+    if (facets !== undefined) facets.original = originalFacets; // needed up read facetChannels
```

Comment:
“needed up read facetChannels” 🤔

Reply:
"needed to reindex the facet channel" (?)
or remove

Reply:
Ah ha. I think I meant “needed **to** read facetChannels”.